### DISCUSSION TITLE
After quitting a Textual app terminal ctrl-c is not working...

### DISCUSSION THREAD

**Discussion by JOJ0** [Textualize/textual#5242] [Q&A] [answered]
...anymore. When I press the key combination, zsh/the terminal/iTerm spits out this: `9;5u`.

Running on macOS Somona, zsh and latest iTerm

Entering `reset` on the terminal then fixes the issue.

I suppose something Textual changes for the control characters is not reset properly when Textual exits. How can I make sure the terminal switches back to its intial state after quitting Textual. Maybe put something in the quit function here:

```
    def action_request_quit(self):
        self.exit(0)
```

Thanks!
Jojo

**Comment by JOJ0** [comment]
For now I worked around the problem with the following. Just execute a reset cli command whenever my Textual app had quit:

```
import atext
import os


def restore_terminal():
    os.system("reset")
...
...
my_textual_app()
atexit.register(restore_terminal)
...
...
```

Is this the usual best practice with Textual apps? Or am I having a specific problem with my terminal/zsh/iTerm configuration ?

**Comment by TomJGooding** [comment]
Textual should take care of restoring the terminal when you exit/quit the app.

What version of Textual are you using? You say it isn't working *anymore* , when did you notice the change?

**Reply by JOJ0** [reply]
Thanks for pointing me there. I think it could be an iTerm issue. I upgraded iTerm recently then it happened. But at the same time I heavily coded in my Textual app, so I thought I might have messed something up in there. It's hard to say which version of iTerm I was running before but I think I had 3.5.7 and went to 3.5.9 (https://github.com/gnachman/iTerm2/tags)

**Comment by willmcgugan** [comment]
I think that's fixed in yesterday's release.

**Reply by JOJ0** [reply]
Thanks, you mean in 0.86.1? Only demo things in here :-) https://github.com/Textualize/textual/compare/v0.86.0...v0.86.1

Or you mean in 0.86.0?

I'm sorry, I never mentioned my version: I'm running 085.2 and in this moment I'm upgrading and will report in a minute....

```
      Successfully uninstalled textual-0.85.2
Successfully installed textual-0.86.1
```

**Reply by JOJ0** [reply, marked as answer]
I removed my workaround and can confirm that the issue is fixed in `0.86.0`. Thanks!
